### ISSUE TITLE
Controller input doesn't work until client PC is rebooted

### ISSUE THREAD

**Issue by Yanitwei** [moonlight-stream/moonlight-qt#1046]
**Describe the bug**
I have a gaming rig that serves as a host pc to stream games from, and I use my work pc as a client. Most of the time I don't power off my work PC so it might be up for a week or two. 

I'm using Xbox Series X/S Wireless Controller (model 1914 to be exact) It doesn't make difference how the controller is connected to client pc (via Bluetooth or via USB). In both cases the controller is properly recognized and the input is working, I can confirm that by diagnostic via "setup USB game controllers" menu or via Steam controller input settings. 
On the host pc though the controller is always recognized as Xbox 360 controller, and I'm checking the same way (game controllers menu) to figure out if it is working or not.

Sometimes I encounter such issue when the input doesn't work even though it is properly recognized on both client and host PC.
In order to fix that I need to restart the client PC. If I restart the host PC though nothing changes.

**Steps to reproduce**
The issue is easily reproduced on its own after couple of uptime days on client pc. Usually the issue re-appears on the next day of a normal remote session (even at the moment I'm writing this report the issue is reproduced)

**Screenshots**
client pc
![client pc game controller](https://github.com/moonlight-stream/moonlight-qt/assets/81416060/7ec904a2-dffd-421b-b8fb-23fa773dc1f1)
host pc
![host pc game controller](https://github.com/moonlight-stream/moonlight-qt/assets/81416060/e48d3cef-0a8d-4633-9c65-e091c00e2778)

**Affected games**
It is not a game specific, when issue appears the controller input simply is not recognized at OS level, I've also checked the input in Steam Big Picture it just doesn't work.

**Other Moonlight clients**
Seems like the issue is only related to my work PC, as at the time when issue appeared on work pc, neither on notebook nor on Android device it doesn't appear.

**Moonlight settings (please complete the following information)**
Mostly default settings, nothing changed. Tried to alter the settings but had no impact, so reverted everything back.

**Gamepad-related issues (please complete if problem is gamepad-related)**
- Do you have any gamepads connected to your host PC directly?
I had a wired Xbox 360 controller always connected to host PC. When I tried to diagnose the issue I disconnected x360 controller from the host and X/S controller on a client started to work again. I thought maybe the issue was related to primary/secondary prioritization as the x360 controller was already connected to host before I've opened a stream on client. After that I haven't connected x360 controller to host again.

- Does the problem still remain if you stream the desktop and use https://html5gamepad.com to test your gamepad?
It still remains, only the triggers (LT/RT) are registered in html5gamepad but with a lag, as if there is some interference.

**Client PC details**
 - OS: Windows 10 22H2
 - Moonlight Version: v4.3.1
 - GPU: Intel HD Graphics 4600

**Server PC details**
 - OS: Windows 10 22H2
 - GeForce Experience version: 3.27.0.112
 - Nvidia GPU driver: 536.40
 - Antivirus and firewall software: Windows Defender and Windows Firewall

**Moonlight Logs (please attach)**
[07 27 not working Moonlight-1690473501.log](https://github.com/moonlight-stream/moonlight-qt/files/12186118/07.27.not.working.Moonlight-1690473501.log)
[07 27 not working Moonlight-1690474581.log](https://github.com/moonlight-stream/moonlight-qt/files/12186119/07.27.not.working.Moonlight-1690474581.log)
[07 27 not working Moonlight-1690475164.log](https://github.com/moonlight-stream/moonlight-qt/files/12186120/07.27.not.working.Moonlight-1690475164.log)
[07 27 not working Moonlight-1690477274.log](https://github.com/moonlight-stream/moonlight-qt/files/12186121/07.27.not.working.Moonlight-1690477274.log)
[07 26 working Moonlight-1690376657.log](https://github.com/moonlight-stream/moonlight-qt/files/12186122/07.26.working.Moonlight-1690376657.log)
[07 26 working Moonlight-1690376953.log](https://github.com/moonlight-stream/moonlight-qt/files/12186123/07.26.working.Moonlight-1690376953.log)

**Comment by cgutman** [comment]
Please try the latest nightly build: https://ci.appveyor.com/project/cgutman/moonlight-qt/builds/47724692/job/djuccbfn2njxw4ik/artifacts

I suspect the SDL2 update may have fixed this.

**Comment by Yanitwei** [comment]
No, unfortunately it behaves exactly the same way as public version.

Attaching logs from moonlight portable x64 r2116
[not working Moonlight-1691576024.log](https://github.com/moonlight-stream/moonlight-qt/files/12301131/not.working.Moonlight-1691576024.log)

[working after restart Moonlight-1691576734.log](https://github.com/moonlight-stream/moonlight-qt/files/12301132/working.after.restart.Moonlight-1691576734.log)

**Comment by Yanitwei** [comment]
No, unfortunately it behaves exactly the same way as public version.

Attaching logs from moonlight portable x64 r2116
[not working Moonlight-1691576024.log](https://github.com/moonlight-stream/moonlight-qt/files/12301131/not.working.Moonlight-1691576024.log)

[working after restart Moonlight-1691576734.log](https://github.com/moonlight-stream/moonlight-qt/files/12301132/working.after.restart.Moonlight-1691576734.log)

**Comment by Solesii** [comment]
Sorry to necro this thread... but is there a fix? This happens to me constantly, especially with FFXIV. Would love a fix in 2024!

**Comment by Yanitwei** [comment]
> Sorry to necro this thread... but is there a fix? This happens to me constantly, especially with FFXIV. Would love a fix in 2024!

Hey 👋,
What moonlight version are you using ? I haven't done anything else but updating moonlight client. Though nothing was mentioned in release notes about this issue, it seem to be fixed on its own. I'm not sure but I do not encounter this issue anymore. 

The origin of the issue could be related to my working pc pretty much as I run different kind of $#it there and constantly change everything.

**Comment by Solesii** [comment]
Hey! ♡

I'm currently running ver. 12.0.2 (beta). I wish updates would fix it. I use Moonlight to stream my games to everything from my TVs to my phone, but almost always I find if I run it for too long, errors arise.

When I return to my PC after closing out moonlight, all my my audio channels return to normal (as they should, obviously lol) but even though my PC clearly recognizes my Xbox One controller, a restart is the only thing that gets games to recognize the inputs again. :(

Again, sorry for rezzing your post & thank you for your response! I still hope a fix will happen eventually! 🙏  (or someone will tell me what I'm doing wrong hopefully lol)

**Comment by Yanitwei** [comment]
> even though my PC clearly recognizes my Xbox One controller, a restart is the only thing that gets games to recognize the inputs again. :(
> 

It seems like your issue is a bit different for you ? In order to resolve the issue do you restart the computer you stream games FROM or the computer you stream games ON ?

**Comment by Solesii** [comment]
The PC *running* the games/GeForce Experience needs restarted to function correctly after streaming. All the devices receiving the streams are fine & aren't the issue. Ty! :)

**Comment by Yanitwei** [comment]
> The PC _running_ the games/GeForce Experience needs restarted to function correctly after streaming. All the devices receiving the streams are fine & aren't the issue. Ty! :)

That is interesting case, a bit different from mine though. In fact, it might not be related to moonlight client at all as it is happening on HOST PC. At this point I'd suggest to try to reproduce the issue while narrowing down the suspects of origin.

Firstly, we need to figure out: When the issue happens does it affect to only specific game(s) or it happens on OS level ? You may confirm it by proceeding to this page https://html5gamepad.com and check the controller input in normal, working scenario. Then, when you have the issue reproduced, proceed again to the same page and check if there's any difference in controller input recognition ?

I know this is simple test and may be annoying but the info we would obtain is very important for further investigations, and if we (or anyone else reading this thread) will come up with a solution in any case this info will be required.

**Comment by Yanitwei** [comment]
> The PC _running_ the games/GeForce Experience needs restarted to function correctly after streaming. All the devices receiving the streams are fine & aren't the issue. Ty! :)

One more thing, try installing "Xbox Extended Feature Support Driver" from Steam settings -> Controller section
![image](https://github.com/moonlight-stream/moonlight-qt/assets/81416060/c2208dd8-59e1-44bf-b3d3-a5cb7e27f211)

**Comment by Solesii** [comment]
Hi! :)  I'll get back to you on the test, but Xbox software & Windows driver software always show controller input even if my game (FFXIV) doesn't. 

Also, I'm not using Steam in this scenario. Causes too many native controller issues at times, so it's not running.

Basically what seems to happen is Nvidia Gamestream (I can't recall specific name, don't judge lol) activates with Moonlight & everything is fine & dandy. Audio channels flip over, inputs work, stream is great. But when I close the Moonlight stream, everything seems to switch back over (ie: audio channels flip back) but any games after stream was ended won't recognize my controller. Even though the driver software, dongle, Xbox accessory software all show the inputs just fine. I've only found restarting the PC to work.

The only odd thing I can think of is that I'm not launching the game directly but rather I load the Desktop through Moonlight & launch my games that way. I wonder if that's causing any issues on the Windows level of things?

(Thank you!)
